src/Microsoft.Health.Fhir.SqlServer.Api/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs

```diff
@@ -103,7 +103,8 @@ internal static void AddSqlServerTableRowParameterGenerators(this IServiceCollec
 
                     if (interfaceType.IsGenericType && interfaceType.GetGenericTypeDefinition() == typeof(ITableValuedParameterRowGenerator<,>))
```

Comment:
Types and interfaces can also be filtered like:
```
bool IsGenerator(Type type)
{
    return type.GetInterfaces().Any(x => x.IsGenericType &&
                                         (x.GetGenericTypeDefinition() == typeof(ITableValuedParameterRowGenerator<,>) ||
                                          x.GetGenericTypeDefinition() == typeof(IStoredProcedureTableValuedParametersGenerator<,>)));
}

serviceCollection.TypesInSameAssemblyAs<SqlServerFhirDataStore>()
    .Where(x => IsGenerator(x.Type))
    .Singleton()
    .AsSelf()
    .AsImplementedInterfaces();
```

Reply:
`TypesInSameAssemblyAs()` filters out open generics. I did add `.Add().Singleton.AsSelf()...` though.